src/hooks/useAsset.tsx

```diff
@@ -0,0 +1,27 @@
+import { useRandomStorageProviderUrl } from '@/api/hooks'
+import { AssetAvailability, DataObject } from '@/api/queries'
+import { createStorageNodeUrl } from '@/utils/asset'
+
+export const useAsset = () => {
+  const randomStorageUrl = useRandomStorageProviderUrl()
+
+  const getAssetUrl = (availability?: AssetAvailability, assetUrls?: string[], dataObject?: DataObject | null) => {
+    if (availability !== AssetAvailability.Accepted) {
+      return
+    }
+    if (assetUrls?.length) {
+      return assetUrls[0]
+    }
+    if (!dataObject?.joystreamContentId) {
+      return
+    }
+    if (dataObject?.liaison?.isActive && dataObject?.liaison?.metadata) {
+      return createStorageNodeUrl(dataObject.joystreamContentId, dataObject?.liaison?.metadata)
+    }
+    if (!dataObject?.liaison?.isActive && randomStorageUrl) {
+      return createStorageNodeUrl(dataObject.joystreamContentId, randomStorageUrl)
+    }
```

Comment:
I don't suppose we will ever hit this but let's add this 

```suggestion
    }
    console.error('No storage provider available to serve asset')
```